### PR TITLE
Basic documentation for run pipeline

### PR DESCRIPTION
**Pull request by adelelopez**
Here's basic documentation to be fleshed out and organized.

### CODE REVIEW

File: doc/fundamentals/run_pipeline.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+In order to get the result of this, you can run `pachctl run pipeline cross-pipe a4 b1 c2`.
+This will create a new job which will create a commit on the pipeline's output branch with the result of that combination of data.
+
+Since a4 is the head of branch A, we could also have done `pachctl run pipeline cross-pipe c2 b1`, which would give the same result.
+Pachyderm is smart enough to automatically use the head for any branches that didn't have a commit specified. 
+Note also that the order you put the commits in doesn't matter, Pachyderm knows how to match them to the right place.
+You also (as always) can reference the head commit of a branch by using the branch name: `pachctl run pipeline cross-pipe A b1 c2`.
+
+This means that if you wanted to run the pipeline on the most recent commits again, you could just run `pachctl run pipeline cross-pipe`.
```
Comment (JoeyZwicker) [inline doc/fundamentals/run_pipeline.md]: How does run-pipeline respect (or not) previously successful datums? In other words, does it have a --reprocess option or do that by default?

File: doc/fundamentals/run_pipeline.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+If you try to use a commit from a branch that is not an input branch, it will give you an error. 
+Similarly, giving it multiple commits from the same branch will give an error.
+
```
Comment (JoeyZwicker) [inline doc/fundamentals/run_pipeline.md]: Will run pipeline just always make a commit on the existing output branch? e.g. It wont break the output commit tree like `update pipeline --reprocess` does?

Comment (adelelopez) [reply doc/fundamentals/run_pipeline.md]: Yes, it will always put it on the existing output branch.

File: doc/fundamentals/run_pipeline.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# Running pipelines on passed commits
+
+Sometimes you need to see the result of merging different commits
+to analyze and identify correct combinations and potential flows in
+data collection and processing. Or, you just want to re-rerun a failed job
```
Comment (adelelopez) [inline doc/fundamentals/run_pipeline.md]: re-rerun -> rerun

**Pull request merged**